Replicator/tests/data/docker/README.md

```diff
@@ -0,0 +1,69 @@
+This docker folder contains the `docker-compose.yml` file for starting and configuring Couchbase Server and Sync Gateways for running non-walrus replicator tests with Sync Gateway.
+
+When running `docker compose up`, there will be one Couchbase Server and two Sync Gateways started as follows:
+
+### 1. Sync Gateway for .SyncServerCollection Tests
+
+ | Config      | Value       |
+ | ----------- | ----------- |
+ | Database    | scratch     |
+ | Port        | 4984        |
+ | Admin Port  | 4985        |
+ | Collections | flowers.roses, flowers.tulips, and flowers.lavenders |
+
+ #### Sync Function
+ ```JS
+    function (doc, oldDoc) { 
+        if (doc.isRejected == 'true') 
+            throw({'forbidden':'read_only'}); 
+        channel(doc.channels); 
+    }
+ ``` 
+
+### 2. Legacy Sync Gateway for .SyncServer30 Tests
+
+ | Config      | Value       |
+ | ----------- | ----------- |
+ | Database    | scratch-30  |
+ | Port        | 4884        |
+ | Admin Port  | 4885        |
+ | Collections | default     |
+
+ #### Sync Function
+
+ Use default sync function which is as follows
+ ```JS
+    function (doc, oldDoc) {
+        channel(doc.channels); 
+    }
+ ``` 
+
+The Admin Credentials of both Sync Gateways are `admin/password` or `Administrtor/password`.
+
+### Docker Compose Environment Variables
+
+The `docker-compose.yml` has 3 environment variables for configuration.
+
+ | Variable      |   Description  |
+ | ------------- | -------------- |
+ | SG_DEB        | Sync Gateway 3.1+ deb file URL. Default is SG 3.1.0 Ubuntu ARM64 URL.      |
+ | SG_LEGACY_DEB | Sync Gateway 3.0 deb file URL. Default is SG 3.0.7 Ubuntu ARM64 URL.       |
+ | SSL           | Boolean flag to configure Sync Gateway for SSL endpoints. Default is true. |
+
+ Note: If you are using Mac x86-64, you must configure SG_DEB and SG_LEGACY_DEB to use x86_64 versions. 
```

Comment:
This can be fairly simply automated.  See https://stackoverflow.com/a/70369688/1155387

Reply:
Currently the URL is set from docker-compose though. Maybe having the env to supply both and docker can pick one based on the architecture.